walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/LinkedDomainsService.kt

```diff
@@ -7,47 +7,53 @@ import com.microsoft.walletlibrary.did.sdk.credential.service.models.linkedDomai
 import com.microsoft.walletlibrary.did.sdk.credential.service.models.linkedDomains.LinkedDomainUnVerified
 import com.microsoft.walletlibrary.did.sdk.credential.service.models.linkedDomains.LinkedDomainVerified
 import com.microsoft.walletlibrary.did.sdk.credential.service.validators.DomainLinkageCredentialValidator
-import com.microsoft.walletlibrary.did.sdk.datasource.network.apis.ApiProvider
+import com.microsoft.walletlibrary.did.sdk.datasource.network.apis.HttpAgentApiProvider
 import com.microsoft.walletlibrary.did.sdk.datasource.network.linkedDomainsOperations.FetchWellKnownConfigDocumentNetworkOperation
 import com.microsoft.walletlibrary.did.sdk.identifier.resolvers.Resolver
 import com.microsoft.walletlibrary.did.sdk.util.Constants
-import com.microsoft.walletlibrary.did.sdk.util.controlflow.Result
-import com.microsoft.walletlibrary.did.sdk.util.controlflow.map
-import com.microsoft.walletlibrary.did.sdk.util.controlflow.runResultTry
+import com.microsoft.walletlibrary.did.sdk.util.controlflow.SdkException
 import com.microsoft.walletlibrary.did.sdk.util.log.SdkLog
 import java.net.URL
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 internal class LinkedDomainsService @Inject constructor(
-    private val apiProvider: ApiProvider,
+    private val apiProvider: HttpAgentApiProvider,
     private val resolver: Resolver,
     private val jwtDomainLinkageCredentialValidator: DomainLinkageCredentialValidator
 ) {
     suspend fun fetchAndVerifyLinkedDomains(relyingPartyDid: String): Result<LinkedDomainResult> {
-        return runResultTry {
-            val domainUrls = getLinkedDomainsFromDid(relyingPartyDid).abortOnError()
-            verifyLinkedDomains(domainUrls, relyingPartyDid)
-        }
+        getLinkedDomainsFromDid(relyingPartyDid)
+            .onSuccess { domainUrls ->
+                return verifyLinkedDomains(domainUrls, relyingPartyDid)
+            }
+            .onFailure {
+                return Result.failure(it)
+            }
+        return Result.failure(SdkException(""))
```

Comment:
Add an exception message that unknown failure happened while fetching/verifying linked domains.